app.js

```diff
@@ -79,7 +95,58 @@ function roomData(req, res, next) {
       returnDataOr404(res, data);
     })
     .catch((err) => {
-      console.log(err)
+      console.log(err);
       next(err);
     });
+}
+
+// Login function, POST request with email and password
+function auth(req, res, next) {
+  const email = req.body.email;
+  const username = req.body.email.split('@')[0]
+  const password = req.body.password;
+  console.log(req.body)
+  if (email && password) {
+    db.oneOrNone(`SELECT pass FROM accounts WHERE email = $1`, [
+      email,
+    ]) // Request the hashed password from the db
+      .then((data) => {
+        if (data) {
+          console.log(data.pass)
+
+          if (bcrypt.compareSync(password, data.pass)) { // compare the hashed password with the login password
+            const accessToken = jwt.sign({ username: username }, accessTokenSecret) // create a signed jwt
+            res.json(
+              {
+                accessToken // respond with the token
+              }
+            );
+          } else {
+
+            res.send('Invalid Login Details!')
+          }
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+        next(err)
+      });
+  } else {
+    res.send('Login Details not recieved!')
+  }
+}
+
+// Creates a new user
+function createUser(req, res, next) {
+  const email = req.body.email;
+  const username = req.body.email.split('@')[0]
```

Comment:
I see here is where you're grabbing the stuff before the @ sign, so I'm assuming we're just storing that...

Reply:
Yes, we are storing that in the database in the username column. Not sure how useful that is because we can just get it from the email but what's done is done.